tests/unit/agent_main_test.py

```diff
@@ -670,3 +673,142 @@ def test_windows_service(self):
             "The service process could not connect to the service controller."
             in err_info.value.stderr.decode()
         )
+
+
+class TestConfigArgumentParsing:
+    def setup(self):
+        self.output_dir = tempfile.mkdtemp()
+        self.output_file = os.path.join(self.output_dir, "config.tar.gz")
+
+        self.config_path = os.path.join(self.output_dir, "agent.json")
+        self.result_dir_path = os.path.join(self.output_dir, "extracted")
+
+        self.extracted_config_path = os.path.join(self.result_dir_path, "agent.json")
+
+    def teardown(self):
+        shutil.rmtree(self.output_dir)
+
+    def _write_config(self, config):
+        with open(self.config_path, "w") as f:
+            f.write(six.ensure_text(json.dumps(config)))
+
+    def _extract_result(self, tar_obj):
+        os.mkdir(self.result_dir_path)
+
+        orig_cwd = os.getcwd()
+        os.chdir(self.result_dir_path)
+        try:
+            tar_obj.extractall()
+        finally:
+            os.chdir(orig_cwd)
+            tar_obj.close()
+
+    def test_export_and_import_config(self):
+        """
+        Test scalyr-agent-3 config --export-config command.
+        """
+
+        original_config = {"api_key": "key"}
+
+        self._write_config(config=original_config)
+
+        subprocess.check_call(
+            [
+                sys.executable,
+                _AGENT_MAIN_PATH,
+                "config",
+                "--export-config",
+                self.output_file,
+                "--config-file",
+                self.config_path,
+            ]
+        )
+
+        tar = tarfile.open(self.output_file)
+        self._extract_result(tar)
+
+        # That same agent config has to be extracted.
+        assert os.path.isfile(self.extracted_config_path)
+
+        with open(self.extracted_config_path, "r") as f:
+            extracted_config = json.load(f)
+
+        assert extracted_config == original_config
```

Comment:
Thanks for adding this check. It's indeed good we assert on the parsed (and also unparsed) content to ensure it doesn't contain some unnecessary `b""`` prefix (which can sometimes end up there in case bytes / string handling is not done correctly in the context of Python 2 / 3).